src/server/src/main/java/io/cassandrareaper/storage/MemoryStorage.java

```diff
@@ -289,7 +289,8 @@ public boolean updateRepairSegment(RepairSegment newRepairSegment) {
   public Optional<RepairSegment> getNextFreeSegmentInRange(UUID runId, Optional<RingRange> range) {
     if (range.isPresent()) {
       for (RepairSegment segment : repairSegmentsByRunId.get(runId).values()) {
-        if (segment.getState() == RepairSegment.State.NOT_STARTED && range.get().encloses(segment.getTokenRange())) {
+        if (segment.getState() == RepairSegment.State.NOT_STARTED
+            && range.get().encloses(segment.getTokenRange().getTokenRanges().get(0))) {
```

Comment:
why only the segment's first token range?

Reply:
That allows the code to be compatible with the new way of storing segments.
Since all token ranges in a segment share the same replicas, working on the master/first one is safe.
Reaper uses the notion of parallel ranges, to separate ranges that can be dealt with concurrently, and if it is safe to run the master range, it is safe to run all other ranges in the segment.

Reply:
same as above, and replaced with `.getBaseRange()`.